### PR TITLE
PMM-8432 Added client-go k8s client

### PR DESCRIPTION
**Pull request by gen1us2k**
PMM-8432

Build: SUBMODULES-2724

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

**Comment by gen1us2k** [comment]
This is the first part of moving from kubectl binary. Tests are missing but it would be great to get feedback from you at this moment.

File: service/k8sclient/internal/kube/kube.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+	cfg.ContentConfig = resource.UnstructuredPlusDefaultContentConfig()
+	cfg.GroupVersion = &gv
+	if len(gv.Group) == 0 {
+		cfg.APIPath = defaultApiUriPath
+	} else {
+		cfg.APIPath = defaultApisUriPath
+	}
+	return rest.RESTClientFor(cfg)
+}
+func (c *Client) Delete(ctx context.Context, rawObj interface{}) error {
```
Comment (gen1us2k) [inline service/k8sclient/internal/kube/kube.go:122]: I think that I need to change rawObj interface{} to runtime.Object, but I'm not sure about it

**Comment by gen1us2k** [comment]
@percona-csalguero @BupycHuk PR is ready for review. Please take a look

**Review by percona-csalguero — APPROVED** [review]

File: service/k8sclient/common/core.go [hunk 2]
```diff
@@ -119,12 +57,18 @@ const (
 )
 
 // IsContainerInState returns true if container is in give state, otherwise false.
-func IsContainerInState(containerStatuses []ContainerStatus, state ContainerState, containerName string) bool {
+func IsContainerInState(containerStatuses []corev1.ContainerStatus, state ContainerState, containerName string) bool {
+	containerState := make(map[string]interface{})
```
Comment (idoqo) [inline service/k8sclient/common/core.go:61]: is there some reason for not making the value here an empty struct? i.e:
```
containerState := make(map[string]struct{})
```

**Review by idoqo — APPROVED** [review]

File: service/k8sclient/common/core.go (hunk 2 above)
Comment (gen1us2k) [reply service/k8sclient/common/core.go:61]: We need it to marshal/unmarshal JSON

**Pull request merged**